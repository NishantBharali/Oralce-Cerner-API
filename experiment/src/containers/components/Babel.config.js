
module.exports = (api) => {
    api.cache(false);
    api.assertVersion('^7.4.4');
    
    const presets = [
    '@babel/preset-env',
    '@babel/preset-react',
    ];
    
    return {
    presets,
    };
    };