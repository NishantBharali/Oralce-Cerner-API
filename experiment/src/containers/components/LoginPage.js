import React from 'react';
import { useState } from 'react';
import { Link, useHistory } from 'react-router-dom';
import axios from 'axios';
import LoginHeader from '../Headers and Footers/LoginHeader';
import FinalFooter from '../Headers and Footers/Footer';
import { Popup } from 'semantic-ui-react';

export const LoginPage = () => {
    
    const [errorMessage, setErrorMessage] = useState('');
    const [username, setUsername] = useState('');
    const [password, setPassword] =useState('');
    const [isValid, setIsValid] = useState(false);
    const [message, setMessage] = useState('');

    const history = useHistory();

    const regEx1 = /[a-zA-Z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,8}(.[a-z{2,8}])?/g;
    
    function checkLogin() {
        //checking localstorage for current user
        if (typeof window !== 'undefined') {
        if(localStorage.getItem('email')){
            history.push('/idealisting')
        }
    }
    }
    checkLogin();

    const usernamelValidation = (e) => {
       
        setUsername(e.target.value);
        if (regEx1.test(username)) {
            setIsValid(true);
          setMessage("");
        } else {
            setIsValid(false);
          setMessage("Enter a valid username.");
        }
      };

    const onLoginClicked = async () => {
        await axios.post('http://localhost:8090/login', {
        username: username,
        password: password
      })
      .then(res => {
        //storing the token and username in the localstorage
        if(res.data.token){
          localStorage.setItem('token', res.data.token)
          localStorage.setItem('email', username)
        }
        history.push('/idealisting')
            })
    .catch(function (error) {
        setErrorMessage("Please check your entries and try again!", error);
        if (typeof window !== 'undefined') {

            localStorage.clear();
            }
    });
    }

    return(
        <div className="content-container">
        <LoginHeader />
            <h1 style={{color:"lightblue", fontSize: "25px", textAlign: "center", fontStyle: "-moz-initial", fontVariant: "all-petite-caps"}}>Log-in to your account</h1>
            {errorMessage && <div className="fail" style={{color: "lightgreen"}}>{errorMessage}</div>}
            <input
                id='input-username'
                pattern='/[a-zA-Z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,8}(.[a-z{2,8}])?/g'
                value={username}
                style={{backgroundColor: 'transparent', border: '1.5px solid white', lineHeight: '20px', color: 'white'}}
                onChange={usernamelValidation}
                className='user icon'
                placeholder="username" />
                <div className={`message ${isValid ? 'success' : 'error'}`} style={{color: 'lightgreen'}}>
                    {message}
                </div>
            <input
                id='input-password'
                type="password"
                style={{backgroundColor: 'transparent', border: '1.5px solid white', lineHeight: '20px', color: 'white'}}
                value={password}
                onChange={e =>setPassword(e.target.value)}
                placeholder="password" />
                <br />

                <button id='btn-primary' className="ui animated button" style={{backgroundColor: 'transparent', border: '3px solid white', color: 'white'}} tableindex="0" 
                    disabled={!username || !password || !isValid}
                    onClick={()=>onLoginClicked()}>
                    <div className="visible content" style={{color: 'lightgrey'}}>Log In</div>
                    <div className="hidden content">
                    <i className="sign in alternate icon" style={{color: 'lightgreen'}}></i></div>
                </button>
                    <hr />
                    <h4 className="mt-2" style={{color: 'lightgrey'}}>
                    &nbsp;&nbsp;&nbsp;&nbsp;Don't have an account? <Link to="/registration" style={{color: 'yellow'}}>&nbsp;Register Here</Link>
                </h4>
        <FinalFooter />
        </div>
    )
}

