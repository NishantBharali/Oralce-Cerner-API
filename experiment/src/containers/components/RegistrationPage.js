import { useState } from 'react';
import { Link, useHistory } from 'react-router-dom';
import validator from 'validator';
import axios from 'axios';
import LoginHeader from '../Headers and Footers/LoginHeader';
import FinalFooter from '../Headers and Footers/Footer';

export const RegistrationPage = () => {
    const [errorMessage, setErrorMessage] = useState('');
    
    const [email, setEmail] = useState('');
    const [password, setPassword] =useState('');
    const [confirmPassword, setConfirmPassword] =useState('');
    const [isValid, setIsValid] = useState(false);
    const [isValid1, setIsValid1] = useState(false);
    const [isValid2, setIsValid2] = useState(false);
    const [message, setMessage] = useState('');
    const [message1, setMessage1] = useState('');
    const [message2, setMessage2] = useState('');
    const history = useHistory();

    const regEx = /[a-zA-Z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,8}(.[a-z{2,8}])?/g;

    const emailValidation = (e) => {
        e.preventDefault();
        setEmail(e.target.value);
        if (regEx.test(email)) {
            setIsValid(true);
          setMessage("Email is valid.");
        } else {
            setIsValid(false);
          setMessage("Please enter a valid email.");
        }
      };

      const passwordValidation = (e) => {
          e.preventDefault();
          setPassword(e.target.value);
          if(validator.isStrongPassword(password, {
              minLength: 8, minLowercase: 1,
              minUppercase: 1, minNumbers: 1, minSymbols: 1
          })) {
              setIsValid1(true);
              setMessage1('This is a strong password.')
          } else {
              setIsValid1(false);
              setMessage1('Minimum 8 characters, an Upper case leter, a numeral and a symbol.')
          }
      };

      const confirmPasswordValidation = (e) => {
        setConfirmPassword(e.target.value);
        if(validator.isStrongPassword(confirmPassword, {
            minLength: 8, minLowercase: 1,
            minUppercase: 1, minNumbers: 1, minSymbols: 1
        })) {
            setIsValid2(true);
            setMessage2('')
        } else {
            setIsValid2(false);
            setMessage2('')
        }
    };

    const onRegisterClicked = async () => {
        await axios.post('http://localhost:8090/user', {
            email: email,
            password: password
      })
      .then(resp => {
        history.push('/');
      })
      .catch(function (error) {
        setErrorMessage("User is already registered!", error);
    });
      
    }

    return(

        <div className="content-container1">
        <LoginHeader />
            <h1 style={{color:"lightblue", fontSize: "25px", textAlign: "center", fontStyle: "-moz-initial", fontVariant: "all-petite-caps"}}>Create an account</h1>
                {errorMessage && <div className="fail" style={{color: "lightgreen"}}>{errorMessage}</div>}
            <input
                id="userName-input"
                value={email}
                style={{backgroundColor: 'transparent', border: '1.7px solid white', lineHeight: '20px', color: 'white'}}
                onChange={emailValidation}
                placeholder="enter your email" />

                <div className={`message ${isValid ? 'success' : 'error'}`} style={{color: 'lightgreen'}}>
                    {message}
                </div>

            <input
                type="password"
                value={password}
                style={{backgroundColor: 'transparent', border: '1.7px solid white', lineHeight: '20px', color: 'white'}}
                onChange={passwordValidation}
                placeholder="enter your password" />

                <div className={`message ${isValid1 ? 'success' : 'error'}`} style={{color: 'lightgreen'}}>
                    {message1}
                </div>

            <input
                type="password"
                value={confirmPassword}
                style={{backgroundColor: 'transparent', border: '1.7px solid white', lineHeight: '20px', color: 'white'}}
                onChange={confirmPasswordValidation}
                placeholder="confirm your password" />

                <div className={`message ${isValid2 ? 'success' : 'error'}`} style={{color: 'lightgreen'}}>
                    {message2}
                </div>

                <hr />
                <button id="btn-register" className="ui animated button" style={{backgroundColor: 'transparent', border: '3px solid white', color: 'white'}} tableindex="0"
                    disabled={!password || !isValid || !isValid1 || !isValid2 || password !== confirmPassword}
                    onClick={()=>onRegisterClicked()}>
                    <div className="visible content" style={{color: 'lightgray'}}>Register</div>
                    <div className="hidden content">
                        <i className="chevron right icon" style={{color: 'lightblue'}}></i>
                    </div>
                </button>

                <hr />
                <h4 className="mt-2" style={{color: 'lightgrey'}}>
                &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Already have an account? <Link to="/" style={{color: 'lightgreen'}}>&nbsp;Login Here</Link>
            </h4>
        <FinalFooter />
        </div>
    )
}
