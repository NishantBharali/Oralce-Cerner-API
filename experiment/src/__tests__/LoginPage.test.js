import React from "react";
import ShallowRenderer from 'react-test-renderer/shallow';
import { configure ,shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import "core-js/stable";
import "regenerator-runtime/runtime";
import { LoginPage } from "../containers/components/LoginPage";

configure({ adapter: new Adapter() });




it("check for test cases",()=> {
    const renderer = new ShallowRenderer();
    renderer.render(<LoginPage />);
    expect(renderer).toMatchSnapshot();

})


const simulateOnChangeInput = (wrapper, inputSelector, newValue) => {
    const input = wrapper.find(inputSelector);
    input.simulate("change", {
      target: { value: newValue },
    });
  
    return wrapper.find(inputSelector);
  };
  test('useState mock ', () => {
    const wrapper = shallow(<LoginPage />);
    const updatedNameInput = simulateOnChangeInput(
        wrapper,
        "#input-password",
        "Test@1728"
      );
    
    wrapper.find('#btn-primary').simulate('click');
    expect(wrapper).toMatchSnapshot();
    // initial state is set and you can now test your component 
    })
    
    test('useState mock for username ', () => {
        const wrapper = shallow(<LoginPage />);
        const updatedName1Input = simulateOnChangeInput(
            wrapper,
            "#input-username",
            "arthur@gmail.com"
          );
        
        wrapper.find('#btn-primary').simulate('click');
        expect(wrapper).toMatchSnapshot();
        // initial state is set and you can now test your component 
        })