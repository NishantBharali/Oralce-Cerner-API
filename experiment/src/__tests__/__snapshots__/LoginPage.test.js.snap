// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check for test cases 1`] = `
ReactShallowRenderer {
  "_context": Object {},
  "_didScheduleRenderPhaseUpdate": false,
  "_dispatcher": Object {
    "readContext": [Function],
    "useCallback": [Function],
    "useContext": [Function],
    "useDebugValue": [Function],
    "useDeferredValue": [Function],
    "useEffect": [Function],
    "useId": [Function],
    "useImperativeHandle": [Function],
    "useInsertionEffect": [Function],
    "useLayoutEffect": [Function],
    "useMemo": [Function],
    "useReducer": [Function],
    "useRef": [Function],
    "useResponder": [Function],
    "useState": [Function],
    "useSyncExternalStore": [Function],
    "useTransition": [Function],
  },
  "_element": <LoginPage />,
  "_firstWorkInProgressHook": Object {
    "memoizedState": "",
    "next": Object {
      "memoizedState": "",
      "next": Object {
        "memoizedState": "",
        "next": Object {
          "memoizedState": false,
          "next": Object {
            "memoizedState": "",
            "next": null,
            "queue": Object {
              "dispatch": [Function],
              "first": null,
            },
          },
          "queue": Object {
            "dispatch": [Function],
            "first": null,
          },
        },
        "queue": Object {
          "dispatch": [Function],
          "first": null,
        },
      },
      "queue": Object {
        "dispatch": [Function],
        "first": null,
      },
    },
    "queue": Object {
      "dispatch": [Function],
      "first": null,
    },
  },
  "_forcedUpdate": false,
  "_idCounter": 0,
  "_instance": null,
  "_isReRender": false,
  "_newState": null,
  "_numberOfReRenders": 0,
  "_renderPhaseUpdates": null,
  "_rendered": <div
    className="content-container"
  >
    <LoginHeader />
    <h1
      style={
        Object {
          "color": "lightblue",
          "fontSize": "25px",
          "fontStyle": "-moz-initial",
          "fontVariant": "all-petite-caps",
          "textAlign": "center",
        }
      }
    >
      Log-in to your account
    </h1>
    
    <input
      className="user icon"
      id="input-username"
      onChange={[Function]}
      pattern="/[a-zA-Z0-9._%+-]+@[a-z0-9.-]+\\\\.[a-z]{2,8}(.[a-z{2,8}])?/g"
      placeholder="username"
      style={
        Object {
          "backgroundColor": "transparent",
          "border": "1.5px solid white",
          "color": "white",
          "lineHeight": "20px",
        }
      }
      value=""
    />
    <div
      className="message error"
      style={
        Object {
          "color": "lightgreen",
        }
      }
    >
      
    </div>
    <input
      id="input-password"
      onChange={[Function]}
      placeholder="password"
      style={
        Object {
          "backgroundColor": "transparent",
          "border": "1.5px solid white",
          "color": "white",
          "lineHeight": "20px",
        }
      }
      type="password"
      value=""
    />
    <br />
    <button
      className="ui animated button"
      disabled={true}
      id="btn-primary"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "transparent",
          "border": "3px solid white",
          "color": "white",
        }
      }
      tableindex="0"
    >
      <div
        className="visible content"
        style={
          Object {
            "color": "lightgrey",
          }
        }
      >
        Log In
      </div>
      <div
        className="hidden content"
      >
        <i
          className="sign in alternate icon"
          style={
            Object {
              "color": "lightgreen",
            }
          }
        />
      </div>
    </button>
    <hr />
    <h4
      className="mt-2"
      style={
        Object {
          "color": "lightgrey",
        }
      }
    >
          Don't have an account? 
      <Link
        style={
          Object {
            "color": "yellow",
          }
        }
        to="/registration"
      >
         Register Here
      </Link>
    </h4>
    <FinalFooter />
  </div>,
  "_rendering": false,
  "_updater": Updater {
    "_callbacks": Array [],
    "_renderer": [Circular],
  },
  "_workInProgressHook": null,
}
`;

exports[`useState mock  1`] = `ShallowWrapper {}`;

exports[`useState mock for username  1`] = `ShallowWrapper {}`;
